src/goose.rs

```diff
@@ -1209,28 +1191,51 @@ impl GooseClient {
             .await;
         let elapsed = started.elapsed();
 
-        // Create a raw request object if we're tracking statistics.
```

Comment:
@LionsAd This is the only downside to this patch -- in order to detect a redirect I have to parse the response -- previously we only did this if stats were enabled. But it doesn't make sense for me for `--sticky-follow` to depend on stats, so now we do it always, whether or not stats are enabled.